addons/payment_payulatam/models/payment_acquirer.py

```diff
@@ -64,7 +64,7 @@ def _payulatam_generate_sign(self, values, incoming=True):
                 self.payulatam_api_key,
                 self.payulatam_merchant_id,
                 values['referenceCode'],
-                float_repr(float(values['amount']), 1),
+                str(values['amount']),
```

Comment:
```suggestion
                float_repr(float(values['amount']), 2),
```
I would choose the other change you suggested, because it is more explicit and it ensures the correct format. 
The value you receive here may not have the expected format, since `amount` is previously computed (in _payment_transaction.py_) with

`'amount': float_repr(processing_values['amount'], self.currency_id.decimal_places or 2)`